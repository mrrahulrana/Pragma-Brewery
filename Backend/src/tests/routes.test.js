const request = require('supertest');

const routes = require('../src/routes');

jest.mock('../src/services/temperatures');

describe('Http server', () => {
  it('should response 404 for unknown paths', async () => {
    const response = await request(routes).get('/');
    expect(response.statusCode).toBe(404);
  });

  it('should respond with a list of beers on /beers path', async () => {
    const response = await request(routes).get('/beers');

    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
