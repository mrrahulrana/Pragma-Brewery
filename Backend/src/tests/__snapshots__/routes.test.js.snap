exports[`Http server should respond with a list of beers on /beers path 1`] = `
Array [
  Object {
    "id": "1",
    "maximumTemperature": 6,
    "minimumTemperature": 4,
    "name": "Pilsner",
    "temperature": 0,
  },
  Object {
    "id": "2",
    "maximumTemperature": 6,
    "minimumTemperature": 5,
    "name": "IPA",
    "temperature": 0,
  },
  Object {
    "id": "3",
    "maximumTemperature": 7,
    "minimumTemperature": 4,
    "name": "Lager",
    "temperature": 0,
  },
  Object {
    "id": "4",
    "maximumTemperature": 8,
    "minimumTemperature": 6,
    "name": "Stout",
    "temperature": 0,
  },
  Object {
    "id": "5",
    "maximumTemperature": 5,
    "minimumTemperature": 3,
    "name": "Wheat beer",
    "temperature": 0,
  },
  Object {
    "id": "6",
    "maximumTemperature": 6,
    "minimumTemperature": 4,
    "name": "Pale Ale",
    "temperature": 0,
  },
]
`;
